codestodians/komlie-frontend
# [Bug] Fix category error on authentication

#### What does this PR do?
- Fixes the disappearance of the category items on logout.
- Delegates cart queries responsibilities to the new `CartContext`
- General refactor

#### How should this be manually tested?
- Authenticate and try to add and remove items from the cart.
- Login and logout and observe that the categories no longer disappear

#### Any background context you want to provide?
- The clearing of the store on logout resulted in the category items disappearance. Also, was noticed that multiple users on the same DB could not properly query for item quantity, and so a task was carried out on the backend to fix that. The other changes on the frontend were done to fix recurring irregularities with observed user actions.

#### What are the relevant pivotal tracker stories?
- [#167557383](https://www.pivotaltracker.com/story/show/167557383)
